aspnetcore/blazor/call-javascript-from-dotnet.md

```diff
@@ -514,11 +514,13 @@ public async ValueTask<string> Prompt(string message)
 }
 ```
 
-## Using JavaScript libraries that render UI (DOM elements)
+## Use of JavaScript libraries that render UI (DOM elements)
 
-Sometimes you may wish to use JavaScript libraries that produce visible user interface elements within the browser DOM. At first glance this might seem difficult because Blazor's diffing system relies on having control over the tree of DOM elements, and will run into errors if some external code mutates the DOM tree and invalidates its mechanism for applying diffs. This is not a Blazor-specific limitation - the same challenge occurs with any diff-based UI framework.
+Sometimes you may wish to use JavaScript libraries that produce visible user interface elements within the browser DOM. At first glance, this might seem difficult because Blazor's diffing system relies on having control over the tree of DOM elements and runs into errors if some external code mutates the DOM tree and invalidates its mechanism for applying diffs. This isn't a Blazor-specific limitation. The same challenge occurs with any diff-based UI framework.
 
-Fortunately it is actually quite simple to embed externally-generated UI within your Blazor component UI reliably. The recommended technique is to have your `.razor` code produce an empty element. As far as Blazor's diffing system is concerned, the element will always be empty, so it doesn't need to recurse into it and will simply leave its contents alone. This makes it safe to populate that element with arbitrary externally-managed content. Here is an example:
+Fortunately, it's straightforward to embed externally-generated UI within a Blazor component UI reliably. The recommended technique is to have the component's code (`.razor` file) produce an empty element. As far as Blazor's diffing system is concerned, the element is always empty, so the component isn't required to recurse into the element and  leave its contents alone. This makes it safe to populate the element with arbitrary externally-managed content.
```

Comment:
One part of this edit looks erroneous:

```
so the component isn't required to recurse into the element and  leave its contents alone
```

This has the opposite meaning than intended. We do want it to leave the contents alone, but this says "it isn't required to leave its contents alone".

```suggestion
Fortunately, it's straightforward to embed externally-generated UI within a Blazor component UI reliably. The recommended technique is to have the component's code (`.razor` file) produce an empty element. As far as Blazor's diffing system is concerned, the element is always empty, so the renderer does not recurse into the element and instead leaves its contents alone. This makes it safe to populate the element with arbitrary externally-managed content.
```